### PR TITLE
Add language kdl

### PR DESCRIPTION
**Pull request by Tamnac**
More info at https://kdl.dev

Node comments (aka "slashdash" comments) are I think technically impossible so they are not implemented, and only comment out the line. Same issue with nested multiline comments

### CODE REVIEW

**Comment by adamharrison** [comment]
Works for me!

File: plugins/language_kdl.lua [hunk 1]
```diff
@@ -0,0 +1,84 @@
+-- mod-version:3
+local syntax = require "core.syntax"
+
+local identifier = "\"?[^%d%s\\/%(%){}<>;%[%]=,\"][^%s\\/%(%){}<>;%[%]=,\"]*\"?"
```
Comment (Guldoman) [inline plugins/language_kdl.lua:4]: What is this matching?

File: plugins/language_kdl.lua [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    { pattern = "[%-+]?0b[01_]+",            type = "number"   },
+    { pattern = "[%-+]?0o[0-7_]+",           type = "number"   },
+    {
+		pattern = "[%-+]?[%d_]+%.[%d_]+e[%-+]?[%d_]+",
+		type = "number"
+	},
+	{ pattern = "[%-+]?[%d_]+%.[%d_]+",      type = "number"   },
+    { pattern = "[%-+]?[%d_]+e[%-+]?[%d_]+", type = "number"   },
+    { pattern = "[%-+]?[%d_]+",              type = "number"   },
+    { pattern = "/[%-/].-\n",                type = "comment"  },
```
[the remark this answers is not in the document]
Comment (Guldoman) [reply plugins/language_kdl.lua:34]: For slashdash and nested multiline comments, you could try something like:
```diff
@@ -3,6 +3,23 @@
 
 local identifier = "\"?[^%d%s\\/%(%){}<>;%[%]=,\"][^%s\\/%(%){}<>;%[%]=,\"]*\"?"
 
+local node_comment = {
+  symbols = {}
+}
+node_comment["patterns"] = {
+  { pattern = { "{", "}" }, type = "comment", syntax = node_comment },
+  { pattern = "[^{}]+", type = "comment" }
+}
+local nested_multiline = {
+  symbols = {}
+}
+nested_multiline["patterns"] = {
+  { pattern = { "/%*", "%*/" }, type = "comment", syntax = nested_multiline },
+  { pattern = "[^/*]+", type = "comment" },
+  { pattern = "%*[^/]", type = "comment" },
+  { pattern = "/[^*]", type = "comment" }
+}
+
 syntax.add {
   name = "KDL",
   files = { "%.kdl" },
@@ -31,8 +48,12 @@
 	{ pattern = "[%-+]?[%d_]+%.[%d_]+",      type = "number"   },
     { pattern = "[%-+]?[%d_]+e[%-+]?[%d_]+", type = "number"   },
     { pattern = "[%-+]?[%d_]+",              type = "number"   },
-    { pattern = "/[%-/].-\n",                type = "comment"  },
-    { pattern = {"/%*", "%*/"},              type = "comment"  },
+    { pattern = { '/%-"', '"' }, type = "comment" },
+    { pattern = { '/%-%s*[a-zA-Z0-9]+%s*=%s*"', '"' }, type = "comment" },
+    { pattern = { "^/%-.*{", "}" }, type = "comment", syntax = node_comment },
+    { pattern = { "/%-{", "}" }, type = "comment", syntax = node_comment },
+    { pattern = "/[%-/].*",                type = "comment"  },
+    { pattern = {"/%*", "%*/"},              type = "comment", syntax = nested_multiline  },
     { pattern = identifier,                  type = "keyword2" },
 	{
 		pattern = "%(()" .. identifier .. "()%)",
```

**Review by Guldoman — COMMENTED** [review]
There is a mix of spaces and tabs in the file.

**Pull request merged**